### ISSUE TITLE
time_zone_select support?

### ISSUE THREAD

**Issue by pfeffer629** [nathanvda/cocoon#479]
Hi,

Is time_zone_select supported? Here is my code:

<%= f.time_zone_select(:time_zone, ActiveSupport::TimeZone.us_zones, { default: Rails.application.config.time_zone }, { class: "" })  %>

In my nested forms, I am returned every time zone, rather than US time zones only. Additionally, my default is set to UTC -0. When I use this same code outside of a nested form, I get the expected response: US time zones, and a selected default.

Thank you for the wonderful gem.

**Comment by nathanvda** [comment]
Weird, I have never used it before but I see no reason why it would not work. If it uses some javascript to activate it, you might have to use the callbacks when adding new items. 

You show very little code, so very hard to say something more, but just guessing: are you sure the form being rendered is the one you are looking at (edit/new/_form) or the partial being rendered. Search if there is another `time_zone_select` in your code? 

I do not see why rendering a form with cocoon, which does nothing to interfere with the rendering of the form except maybe for the dynamically added items, would suddenly "forget" constants.

**Comment by pfeffer629** [comment]
Hi Nathan,

Sorry for the late response. I've tried hard coding the default as a string like so:

<%= f.time_zone_select(:time_zone, ActiveSupport::TimeZone.us_zones, { default: "Mountain Time (US & Canada)" })  %>

In addition, the same exact line of code performs as expected when used outside of a nested form (even in the parent form).

Here's some more code for reference:

Parent form:

```
<%= form_for(@transportation_logistic, url: client_travel_plan_transportation_logistics_path, html: {class: 'white-placeholder'}) do |f| %>
<div class="clearfix nested-rows">
    <%= f.fields_for :transportation_segments do |transportation_segment| %>
      <%= render 'client/transportation_logistics/partials/transportation_segment', f: transportation_segment %>
    <% end %>
  </div>
  <div>
    <%= link_to_add_association "Add Another Transportation Segment", f, :transportation_segments, { partial: "client/transportation_logistics/partials/transportation_segment", class: "white-link pull-right" } %>
</div>

```

I've tried deleting every field except for the time zone input. Another thing to note is that the dropdown works exactly as intended except for setting my default.

**Comment by nathanvda** [comment]
Ok, weird, in one project it just worked (rails 4), in another the default was not set (rails 5). Which rails version are you using?  

In the current version I found the following offender https://github.com/rails/rails/blob/6c4a32e27ddadc58673a36f47ca28f74b09c9ed0/actionview/lib/action_view/helpers/tags/time_zone_select.rb#L16 

which has changed when compared to the rails 4 version. In rails 4 they wrote: 

    value(@object) || @options[:default]

and now it is just 
   
    value || @options[:default]

and I noticed that for the string fields I was using, from the database, the "value" was never actually nil, just an empty string (which was not what I expected to be honest). 

As a work-around you could set the time-zone before rendering (if it was not set?). Something like `f.object.time_zone ||= 'Mountain Time (US & Canada)'` should work? And you could use the `wrap_object` option to do it for your nested items. 

E.g. something like 

```
<%= link_to_add_association "Add Another Transportation Segment", f, :transportation_segments, 
             partial: "client/transportation_logistics/partials/transportation_segment", 
             class: "white-link pull-right",
             wrap_object: Proc.new {|ts| ts.time_zone ||= 'Brussels' }) 
 %>
```

**Comment by pfeffer629** [comment]
I am currently using Rails 4.

Apologies, I'm not familiar with wrap_object. I copied this line of code, but it didn't do anything.

`wrap_object: Proc.new {|ts| ts.time_zone ||= 'Brussels' }) `

For further clarification on my end, the default is automatically set to UTC no matter what. When I edit an object, the time zone dropdown populates with the correct time zone. I've tried setting the value explicitly with both:

`value: f.object.time_zone ||= 'Mountain Time (US & Canada)'`
`selected: f.object.time_zone ||= 'Mountain Time (US & Canada)'`

Either way still shows up as UTC
